tikv-client/src/main/java/com/pingcap/tikv/allocator/RowIDAllocator.java

```diff
@@ -200,8 +217,10 @@ private long updateHash(
       return 0L;
     }
 
-    set(dataKey, newVal);
-    updateMeta(key, oldVal, snapshot);
+    ArrayList<BytePairWrapper> pairs = new ArrayList<>();
```

Comment:
You can create ArrayList with initialCapacity as two explicitly. Otherwise it creates an array that's larger than necessary.